src/Microsoft.OData.Edm/Csdl/Serialization/EdmModelCsdlSerializationVisitor.cs

```diff
@@ -691,7 +691,15 @@ private void VisitPropertyRefs(IEnumerable<IEdmStructuralProperty> properties)
         {
             foreach (IEdmStructuralProperty property in properties)
             {
-                this.schemaWriter.WritePropertyRefElement(property);
+                var prop = property as IStructuredKeyAlias;
+                if (string.IsNullOrEmpty(prop?.KeyAlias))
+                {
+                    this.schemaWriter.WritePropertyRefElement(property);
+                }
+                else 
+                {
+                    this.schemaWriter.WritePropertyRefElement(prop.Path, prop.KeyAlias);

```

Comment:
Maybe this is asserted by the caller somewhere, but can't `prop` still potentially be `null` at this point?

Reply:
am checking if the prop is null in the if--statement.